### ISSUE TITLE
add ability to ensure cloned satellite is isolated from rest of environment

### ISSUE THREAD

**Issue by beav** [RedHatSatellite/satellite-clone#134]
If you are using satellite-clone for testing upgrades, it is important that the cloned satellite is not able to talk to any other components of the production environment. The main thing to watch out for would be capsules but there might be other things as well.

Ideally, satellite-clone would have a flag to either block communication or remove any references to external services like capsules.

**Comment by johnpmitsch** [comment]
I think we should remove all capsules from the database after a clone as they won't stay functional and are going to create all kinds of connection errors during regular operation and upgrade.

**Comment by sthirugn** [comment]
This should be a blocker for 1.1.0 as we don't want more users to start using this and messing up existing capsules.

**Comment by johnpmitsch** [comment]
clarification on irc convo:

```
     beav | so when i clone, i want to be 100% guaranteed sure that its not going to ever talk to a capsule
     beav | that would be super bad
     beav | so i looked at the upgrade code and didnt see it talk to capsules which is good
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
     beav | but if there was some other way to ensure it never does that, that would be better
 jomitsch | what do you mean by upgrade code?
     beav | oh like satellite-installer                                                                                  
     beav | like in the pre/post hooks  
 jomitsch | did you use sat-clone at all?                                                                     
     beav | like, if i cloned a satellite, i think if i kicked off a sync, it would tell the capsule to sync  
 jomitsch | In our small testing, we've found that not to be the case                                         
 jomitsch | but def is soemthing we want to double check
     beav | ya, that was kinda my issue. we ended up just cloning by hand, the customer asked me if it was going to talk
          | to capsules and i wasnt 100% sure                                                                           
     beav | i was like "i dont think it will" but that was not sufficient :)
 jomitsch | lol, yes, thats something we should check and clarify one way or the other                        
     beav | ya exactly:)                                                                                      
 jomitsch | and then if they do talk to the capsules, make sure they don't
     beav | so that is the reason i put in the feature request really                                         
     beav | so i could be at a customer site and be 100% confident
```

**Comment by johnpmitsch** [comment]
TL;DR - we should test and then clarify if a cloned satellite will talk to the existing capsules. If it turns out a cloned sat does communicate with existing capsules, then we wan't to prevent this behavior. If this is not the case, then we should make this clear to users.

Moving to 1.1.1 since its not a blocker

**Comment by sthirugn** [comment]
`If this is not the case, then we should make this clear to users.` - I disagree, if this is not the case, this should be fixed, otherwise the cloned satellite cannot be used without affecting production capsules.

**Comment by johnpmitsch** [comment]
@sthirugn I think we are on the same page :smile: what I am saying is:
cloned sats do talk to the capsule - we should prevent this behavior and fix it
cloned sats do *not* talk to the capsule - we should make this clear in the docs

I'm pretty sure we found that cloned sats do not talk to existing capsules, but would want to double check before making any decisions based on this.

**Comment by sthirugn** [comment]
@johnpmitsch I got confused since this is moved to 1.1.1 milestone. ty for the clarification.

**Comment by johnpmitsch** [comment]
I tested this out:
- spun up a satellite and capsule with content, and ran a capsule sync
- ran katello-backup
- cloned to a new machine using the backup
- Using the cloned Satellite, I tried to kick off a capsule sync with the capsule from the original Satellite.
- Received an error "getaddrinfo: Name or service not known"

 The cloned Satellite was not able to talk to the original Satellite.

A thought for the future is - Do we automatically want to delete external Capsules and Clients on a cloned Satellite? This could avoid alot of errors when promoting CVs and such.

**Comment by sthirugn** [comment]
Did you kick off capsule sync from Capsule repos page? If so, can you also check:
- Sync a repo in original satellite, associate it to a capsule. Ensure capsule sync is completed.
- Clone the satellite
- Re-sync the repo in cloned satellite and check if this triggers sync in the capsule?

Please create a separate issue for deleting external capsules and clients.  If an existing satellite server is crashed and user uses satellite-clone to standup a new server, he will lose all info about capsules/clients :/

**Comment by johnpmitsch** [comment]
@sthirugn I will try that, but in katello the same capsule sync is kicked off whether its from the capsule page, after a CV promotion, or a repo sync in library, so I'm pretty confident it will be the same result

**Comment by beav** [comment]
@johnpmitsch how did it get `name or service not known` error? Typically, the capsule's hostname is resolvable from the cloned satellite.

**Comment by johnpmitsch** [comment]
With further investigation, we found that a cloned satellite can still talk to a capsule if the capsule hostname can be resolved by the cloned satellite. The existing capsule still talks to the main satellite. We came up with the solution of dissociating the capsules from any Lifecycle environments after a clone. Post-clone behavior of clients has to be investigated as well.

@beav @sthirugn feel free to clarify/correct any points
